src/serverModel.ts

```diff
@@ -153,28 +304,37 @@ export default function ServerModel(mpInstance) {
             };
         }
 
-        return jsonObject;
+        return jsonObject as IConsentStateDTO;
     };
 
-    this.createEventObject = function(event, user) {
-        var uploadObject = {};
-        var eventObject = {};
+    this.createEventObject = function(
+        event: BaseEvent,
+        user?: MParticleUser
+    ): SDKEvent | IUploadObject {
+        var uploadObject: Partial<IUploadObject> = {};
+        var eventObject: Partial<SDKEvent> = {};
 
+        // TODO: What is the purpose of setting OptOut to the inverse of the Store's
+        //       isEnabled property and not just `false`?
         var optOut =
             event.messageType === Types.MessageType.OptOut
                 ? !mpInstance._Store.isEnabled
                 : null;
 
+        // TODO: Why is Webview Bridge Enabled or Opt Out necessary here?
         if (
             mpInstance._Store.sessionId ||
-            event.messageType == Types.MessageType.OptOut ||
+            event.messageType === Types.MessageType.OptOut ||
             mpInstance._Store.webviewBridgeEnabled
         ) {
             if (event.hasOwnProperty('toEventAPIObject')) {
                 eventObject = event.toEventAPIObject();
             } else {
                 eventObject = {
-                    EventName: event.name || event.messageType,
+                    // TODO: Do we need the number from the enum or the string it evaluates to?
```

Comment:
This is a quirk of the backend I believe for V2 - it should be the number from the enum.  For v2 of the events endpoint, the name for Session Start/End, AST's are the numbers (1/2 and 10).  For V3 this resolves back to `session_start/end`, and `application_state_transition`.